blssig/aggregation.go

```diff
@@ -46,19 +49,50 @@ func (v Verifier) VerifyAggregate(msg []byte, signature []byte, pubkeys []gpbft.
 	return v.scheme.Verify(aggPubKey, msg, signature)
 }
 
-func (v Verifier) pubkeysToMask(pubkeys []gpbft.PubKey) (*sign.Mask, error) {
+func (v *Verifier) pubkeyToPoint(p gpbft.PubKey) (kyber.Point, error) {
+	if len(p) > 96 {
+		return nil, fmt.Errorf("public key is too large: %d > 96", len(p))
+	}
+
+	v.mu.RLock()
+	point, ok := v.pointCache[string(p)]
+	v.mu.RUnlock()
+	if ok {
+		return point.Clone(), nil
+	}
+
+	point = v.keyGroup.Point()
+	err := point.UnmarshalBinary(p)
+	if err != nil {
+		return nil, fmt.Errorf("unarshalling pubkey: %w", err)
+	}
+	if point.Equal(v.keyGroup.Point().Null()) {
+		return nil, fmt.Errorf("public key is a null point")
+	}
+	v.mu.Lock()
+
+	// Initialize the cache, or re-initialize it if we've grown too big. We don't expect the
+	// latter to happen in practice (would need over 10k participants), but better be safe than
+	// sorry. We could, alternatively, use an LRU but... that's not worth the overhead for
+	// somethign that shouldn't happen.
+	if v.pointCache == nil || len(v.pointCache) >= maxPointCacheSize {
+		v.pointCache = make(map[string]kyber.Point)
+	}
+
+	v.pointCache[string(p)] = point
+	v.mu.Unlock()
+
+	return point.Clone(), nil
```

Comment:
Dumb question: `point.Clone` is safe for concurrent use; correct?

Reply:
It's behind an interface but... if it's not, something is very wrong with kyber. Honestly, I'm not even sure if the clone is necessary, I'm just trying to be extra safe.